### DISCUSSION TITLE
Never upgrade the Perl version

### DISCUSSION THREAD

**Discussion by dolmen** [renovatebot/renovate#23472] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I am trying to get this working for the first time

### Describe the problem

*TL;DR*: Upgrading the Perl version must just be left to the project maintainers.

The @renovate-bot sends PRs to upgrade the Perl version in `cpanfile` configuration to the latest Perl version just released a few days ago.
Example: https://github.com/cucumber/gherkin/pull/140

This is completely WRONG because:
* The Perl version is the requirement MINIMUM version of the Perl runtime. Requiring the latest version of the runtime which is not yet available in Linux distribution repositories just means that nobody will be able to use the project anymore, including CI environments (ex: GitHub actions ubuntu-latest). So a perl module that only requires `v5.8.8` (released 17 years ago) is just fine (the Perl community has a strong sense of backward compatibility).
* Upgrading `v5.38.0` -> `v5.39.0` is a MAJOR upgrade in the perl world. For example this is where deprecations happen, or when features disappear. The `v5` is not significant as `v5` will stay forever (`v4` -> `v5` happened 30 years ago; `v6` is now called [Raku](https://raku.org/), `v7` [will not happen](https://blogs.perl.org/users/psc/2022/05/what-happened-to-perl-7.html)).

Note: In general, do not rely on SEMVER semantics for Perl modules: Perl module distribution has existed long before SEMVER has been specified, and most modules do not follow SEMVER semantics.

### Have you created a minimal reproduction repository?

I have linked to a minimal reproduction in the description above

**Comment by rarkins** [comment, marked as answer]
It's not wrong, it's not a problem, and ignoring/disabling dependencies is clearly documented

**Reply by dolmen** [reply]
Ok.

I get it. This project is driven by bad incentives: make tons of PRs.

Usefulness to maintainers by using sane default settings tailored to each programming community is far below.

**Reply by rarkins** [reply]
This user has been blocked for inflammatory comments - we have a low tolerance for such negativity. Unblocking is done as generously as blocking though - a simple email is enough.

**Comment by rarkins** [comment]
I’m happy to discuss a change in semantics for Perl updating with anyone as long as it’s done with some politeness. It sounds similar to the go module directive behavior we implemented.

Please start a new discussion if so.
